js/modules/k6/http/request.go

```diff
@@ -73,6 +73,61 @@ func (c *Client) Request(method string, url goja.Value, args ...goja.Value) (*Re
 	return c.responseFromHTTPext(resp), nil
 }
 
+// Request makes an http request of the provided `method` and returns a corresponding response by
+// taking goja.Values as arguments
+func (c *Client) asyncRequest(method string, url goja.Value, args ...goja.Value) (*goja.Promise, error) {
+	state := c.moduleInstance.vu.State()
+	if state == nil {
+		return nil, ErrHTTPForbiddenInInitContext
+	}
+
+	var body interface{}
+	var params goja.Value
+
+	if len(args) > 0 {
+		body = args[0].Export()
+	}
+	if len(args) > 1 {
+		params = args[1]
+	}
+
+	rt := c.moduleInstance.vu.Runtime()
+	req, err := c.parseRequest(method, url, body, params)
+	if err != nil {
+		if state.Options.Throw.Bool {
+			return nil, err
+		}
+		state.Logger.WithField("error", err).Warn("Request Failed")
+		r := httpext.NewResponse()
+		r.Error = err.Error()
+		var k6e httpext.K6Error
+		if errors.As(err, &k6e) {
+			r.ErrorCode = int(k6e.Code)
+		}
+		p, resolve, _ := rt.NewPromise()
+		resolve(&Response{Response: r, client: c})
+		return p, nil
+	}
+
+	p, resolve, reject := rt.NewPromise()
+	callback := c.moduleInstance.vu.RegisterCallback()
+
+	go func() {
+		resp, err := httpext.MakeRequest(c.moduleInstance.vu.Context(), state, req)
+		callback(func() error {
+			if err != nil {
+				_ = reject
```

Comment:
I'm unclear as to what this statement does. Could you elaborate? Also, how does it relate to returning the error? 🙇🏻

Reply:
Ah... good catch. 

The line just tells go to not care about the fact that I do not use `reject`. Which is as I wasn't sure whether I should be rejecting this promise or just returning an error. But thinking about it ... this will in practice always stop the event loop which is definitely not what should happen so I will have to make a test for this and actually check that I don't break the event loop ;) 

Connected to https://github.com/grafana/k6/pull/2877/files#r1089241147

good catch again :bow:

Reply:
Hehe, I'm glad I helped catch a potential issue... somehow 😄 🙇🏻

Reply:
fixed in https://github.com/grafana/k6/pull/2877/commits/9d6afbc5a2a9420d1cfacc63b953a21ebe23989a